common/defs.go

```diff
@@ -21,10 +21,19 @@ import (
 	"fmt"
 	"net"
 	"strings"
+	"encoding/json"
 )
 
 // Here we only keep type definitions and struct definitions with no behavior.
 
+func String(i interface{}) string {
```

Comment:
Is 'String' really the most descriptive name we can give this function? The name is very ... generic and doesn't really tell us what it does.